apiserver/facades/controller/caasapplicationprovisioner/provisioner.go

```diff
@@ -167,7 +168,7 @@ func NewStateCAASApplicationProvisionerAPI(stdCtx context.Context, ctx facade.Mo
 	)
 
 	apiGroup := &APIGroup{
-		PasswordChanger:    common.NewPasswordChanger(st, common.AuthFuncForTagKind(names.ApplicationTagKind)),
+		PasswordChanger:    common.NewPasswordChanger(passwordService, st, common.AuthFuncForTagKind(names.ApplicationTagKind)),
```

Comment:
If this lands before #19329 I'll move it to the caasapplicationprovisoner worker directly.

FYI: We need this to set the password of the application. For k8s we use the application tag for authentication of a call to the `caasapplication` `UnitIntroduction` facade method. The only setter is the `caasapplicationprovisioner` worker. Ideally we would completely drop authenticating as the application "agent", but shortest path is to continue supporting it only for this functionality.